compiler/src/dotty/tools/dotc/transform/ReifyQuotes.scala

```diff
@@ -625,6 +629,9 @@ class ReifyQuotes extends MacroTransformWithImplicits with InfoTransformer {
       }
     }
 
+    private def isStage0Value(sym: Symbol)(implicit ctx: Context): Boolean =
+      sym.is(Inline) && sym.owner.is(Macro) && !defn.isFunctionType(sym.info)
```

Comment:
The addition of `!defn.isFunctionType(sym.info)` is the only real change in this PR, the rest is refactoring.

Reply:
LGTM

Reply:
After discussion with Aggelos, also LGTM for me. 

@Blaisorblade "cross stage persistence in reverse" is a good way to characterize this.